include/nix/DataFrame.hpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2017, German Neuroinformatics Node (G-Node)
+//
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted under the terms of the BSD License. See
+// LICENSE file in the root of the Project.
+//
+// Author: Christian Kellner <ckellner@redhat.com>
+
+#ifndef NIX_DATA_FRAME_H
+#define NIX_DATA_FRAME_H
+
+#include <nix/Platform.hpp>
+#include <nix/DataType.hpp>
+
+#include <nix/base/EntityWithSources.hpp>
+#include <nix/base/IDataFrame.hpp>
+
+#include <nix/Hydra.hpp>
+
+#include <string>
+#include <vector>
+
+namespace nix {
+
+class NIXAPI DataFrame : public base::EntityWithSources<base::IDataFrame> {
+public:
+
+    DataFrame()
+        : EntityWithSources()
+        {}
+
+    DataFrame(const std::shared_ptr<base::IDataFrame> &df)
+        : EntityWithSources(df)
+        {}
+
+    DataFrame(std::shared_ptr<base::IDataFrame> &&ptr)
+        : EntityWithSources(std::move(ptr))
+        {}
+
+    ndsize_t rows() const {
+        return backend()->rows();
+    }
+
+    void rows(ndsize_t n) {
+        return backend()->rows(n);
+    }
+
+    virtual std::vector<unsigned> nameToCol(const std::vector<std::string> &names) const {
+        return backend()->nameToCol(names);
+    }
+
+    virtual std::vector<std::string> colToName(const std::vector<unsigned> &cols) const {
+        return backend()->colToName(cols);
+    }
+
+    std::vector<Column> columns() const {
+        return backend()->columns();
+    }
+
+    void writeRow(ndsize_t row, const std::vector<Variant> &v) {
+        return backend()->writeRow(row, v);
+    }
+
+    void writeCell(ndsize_t row, unsigned col, const Variant &v) {
+        return this->writeCells(row, {{col, v}});
+    }
+
+    void writeCells(ndsize_t row, const std::vector<Cell> &cells) {
+        return backend()->writeCells(row, cells);
+    }
+
+    Variant readCell(ndsize_t row, unsigned col) const {
+        const std::vector<std::string> &cols = colToName({col});
+        if (cols.size() < 1) {
+            return Variant{};
+        }
+        return this->readCell(row, cols[0]);
+    }
+
+    Variant readCell(ndsize_t row, const std::string &col) const {
+        std::vector<Variant> cells = backend()->readCells(row, {col});
+        if (cells.size() < 1) {
+            return Variant{};
+        }
+
+        return cells[0];
+    }
+
```

Comment:
would it make sense to offer a function to get the actual ``Cell`` (``vector<Cell> getCell(row, col)``)object?

Reply:
I thought maybe `getCells(std::vector<Cell> &cells, ndsize_t row, std::vector<unsgined> cols, gboolean resize_vector = true)` ... but one that returns a new vector of `Cell`s could also work.

Reply:
dunno, both versions seem fine to , except that returning of a new vector of Cells does not imply that the content is a reference to the actually stored cells...

Reply:
Maybe `Cell` should inherit from `Variant` and then all `getCell(s)` methods could just return `Cells`.